plugins/permCheck.ts

```diff
@@ -19,21 +27,247 @@
  * @end
  */
 
-import type { GuildMember, PermissionResolvable } from "discord.js";
-import { CommandControlPlugin, CommandType, controller } from "@sern/handler";
-export function permCheck(perm: PermissionResolvable, response: string) {
-	return CommandControlPlugin<CommandType.Both>(async (ctx, args) => {
-		if (ctx.guild === null) {
-			await ctx.reply("This command cannot be used here");
-			console.warn(
-				"PermCheck > A command stopped because we couldn't check a users permissions (was used in dms)",
-			); //delete this line if you dont want to be notified when a command is used outside of a guild/server
-			return controller.stop();
-		}
-		if (!(ctx.member! as GuildMember).permissions.has(perm)) {
-			await ctx.reply(response);
-			return controller.stop();
-		}
-		return controller.next();
-	});
+import {
+  Colors,
+  EmbedBuilder,
+  PermissionsBitField,
+  type GuildMember,
+  type PermissionResolvable,
+  type TextChannel
+} from "discord.js";
+import {
+  CommandControlPlugin,
+  CommandType,
+  controller,
+  Service
+} from "@sern/handler";
+
+function command(perm: PermissionResolvable, response?: string) {
+  return CommandControlPlugin<CommandType.Both>(async (ctx) => {
+    if (ctx.guild === null) {
+      await ctx.reply("This command cannot be used in DM's!");
+      return controller.stop();
+    }
+    if (
+      !(ctx.member! as GuildMember)
+        .permissionsIn(ctx.channel as TextChannel)
+        .has(perm)
+    ) {
+      await ctx.reply({
+        embeds: [
+          sendEmbed(
+            response ??
+              `You are missing required permissions to run this command:\n${permsToString(perm)}`
+          )
+        ],
+        ephemeral: ctx.isMessage() ? false : true
```

Comment:
`!ctx.isMessage()` *should* return the opposite boolean

Reply:
Yes I will make that change actually.